lisa/energy_model.py

```diff
@@ -538,7 +538,12 @@ def _guess_freqs(self, cpu_utils):
         ideal_freqs = [0 for _ in self.cpus]
         for node in self.cpu_nodes:
             [cpu] = node.cpus
-            required_cap = cpu_utils[cpu]
+
+            # At least 25% of capacity margin should be provided to meet the
+            # demand at a given utilizaton level as per the scheduler's
+            # 'capacity_margin' coefficient and schedutil's map_util_freq(),
+            # so reflect that here
+            required_cap = cpu_utils[cpu] * 1.25
```

Comment:
Maybe I don't have my thoughts straight, but aren't we mixing two concepts here?

One is the `capacity_margin` (20% default) which determines the overutilized threshold, and which we use in `task_fits_capacity()`.

The other is `map_util_freq()` which relates to schedutil trying to give us a frequency 25% higher than what the `utilization` requires. So IIUC with this we're using `map_util_freq()` for capacity selection, which applies to both CPU and frequencies.

In the end I think my rambling is that we don't actually see `capacity_margin` here.

Reply:
Well IMO both are exactly the same thing. At some point I actually wanted to send a patch that makes sugov use `capacity_margin` directly. And I still think we should do that. Both for capacity awareness and frequency selection we try to have at least 20% of idle time.